### PR TITLE
EdkRepo: Create unit tests for the class BaseXmlHelper() and modulari…

### PR DESCRIPTION
**Pull request by VictorMlVillarreal**
…ze its existing methods if needed

Implemented unit tests and documentation for the BaseXmlHelper() class to improve clarity, reliability, and maintainability. Some internal methods were refactored to extract logic into smaller helper functions, making the codebase easier to test.

Changes:
-Added unit tests for the BaseXmlHelper() class, covering the main execution paths and edge cases. 
-Created the corresponding documentation file BaseXmlHelper_TestCases.md. 
-Extracted logic from __init__(self, fileref, xml_types) into a dedicated helper function called _load_tree(fileref, json_to_xml_fn). 
-All new tests rely on mocked dependencies and do not perform filesystem or network operations. 
-Updated internal references to use the refactored helpers.

Fixes: #326

### CODE REVIEW

File: edkrepo_manifest_parser/edk_manifest.py [hunk 1]
```diff
@@ -129,12 +120,26 @@ def _build_etree_node(self, current_dict, parent):
...
+    def _load_tree(self, fileref):
+        """Load an ElementTree from a .xml file or a .json file via self._json_to_xml; raise TypeError if the file is invalid or the extension is unsupported."""
+        try:
+            file_ext = os.path.splitext(fileref)[1]
+            if file_ext == '.xml':
+                return ET.ElementTree(file=fileref)  # fileref can be a filename or filestream
+            elif file_ext == '.json':
+                return self._json_to_xml(fileref)
+            else:
+                raise TypeError(INVALID_XML_ERROR.format(fileref, et_error))
```
Comment (ashedesimone) [inline edkrepo_manifest_parser/edk_manifest.py]: et_error does not exist until the exception clause. I suggest that we change this to:

```
 raise TypeError(INVALID_XML_ERROR.format(fileref, 'Unsupported file extension:'.format(file_ext'))
```
(define the string with other string constants for the file)

**Review by ashedesimone — CHANGES_REQUESTED** [review]

**Review by kevinsun49 — APPROVED** [review]
